File: airbyte-config/config-persistence/src/test/java/io/airbyte/config/persistence/HealthCheckPersistenceTest.java
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.config.persistence;

import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;
import static org.mockito.ArgumentMatchers.any;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

import io.airbyte.data.services.impls.jooq.ActorDefinitionServiceJooqImpl;
import io.airbyte.data.services.impls.jooq.CatalogServiceJooqImpl;
import io.airbyte.data.services.impls.jooq.ConnectionServiceJooqImpl;
import io.airbyte.data.services.impls.jooq.ConnectorBuilderServiceJooqImpl;
import io.airbyte.data.services.impls.jooq.DestinationServiceJooqImpl;
import io.airbyte.data.services.impls.jooq.HealthCheckServiceJooqImpl;
import io.airbyte.data.services.impls.jooq.OAuthServiceJooqImpl;
import io.airbyte.data.services.impls.jooq.OperationServiceJooqImpl;
import io.airbyte.data.services.impls.jooq.OrganizationServiceJooqImpl;
import io.airbyte.data.services.impls.jooq.SourceServiceJooqImpl;
import io.airbyte.data.services.impls.jooq.WorkspaceServiceJooqImpl;
import io.airbyte.db.Database;
import java.sql.SQLException;
import org.jooq.Result;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

class HealthCheckPersistenceTest {

  private Database database;
  private ConfigRepository configRepository;

  @BeforeEach
  void beforeEach() throws Exception {

    database = mock(Database.class);
    configRepository = new ConfigRepository(
        new ActorDefinitionServiceJooqImpl(database),
        new CatalogServiceJooqImpl(database),
        new ConnectionServiceJooqImpl(database),
        new ConnectorBuilderServiceJooqImpl(database),
        new DestinationServiceJooqImpl(database),
        new HealthCheckServiceJooqImpl(database),
        new OAuthServiceJooqImpl(database),
        new OperationServiceJooqImpl(database),
        new OrganizationServiceJooqImpl(database),
        new SourceServiceJooqImpl(database),
        new WorkspaceServiceJooqImpl(database));
  }

  @Test
  void testHealthCheckSuccess() throws SQLException {
    final var mResult = mock(Result.class);
    when(database.query(any())).thenReturn(mResult);

    final var check = configRepository.healthCheck();
    assertTrue(check);
  }

  @Test
  void testHealthCheckFailure() throws SQLException {
    when(database.query(any())).thenThrow(RuntimeException.class);

    final var check = configRepository.healthCheck();
    assertFalse(check);
  }

}

File: airbyte-config/config-persistence/src/test/java/io/airbyte/config/persistence/HealthCheckPersistenceTest.kt
/*
 * Copyright (c) 2020-2025 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.config.persistence

import io.airbyte.data.services.HealthCheckService
import io.airbyte.data.services.impls.jooq.HealthCheckServiceJooqImpl
import io.airbyte.db.ContextQueryFunction
import io.airbyte.db.Database
import org.jooq.Result
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.mockito.Mockito
import java.sql.SQLException

internal class HealthCheckPersistenceTest {
  private var database: Database? = null
  private var healthCheckService: HealthCheckService? = null

  @BeforeEach
  @Throws(Exception::class)
  fun beforeEach() {
    database = Mockito.mock<Database>(Database::class.java)
    healthCheckService = HealthCheckServiceJooqImpl(database)
  }

  @Test
  @Throws(SQLException::class)
  fun testHealthCheckSuccess() {
    val mResult = Mockito.mock<Result<*>?>(Result::class.java)
    Mockito.`when`<Any?>(database!!.query<Any?>(org.mockito.kotlin.any<ContextQueryFunction<Any?>>())).thenReturn(mResult)

    val check = healthCheckService!!.healthCheck()
    Assertions.assertTrue(check)
  }

  @Test
  @Throws(SQLException::class)
  fun testHealthCheckFailure() {
    Mockito.`when`<Any?>(database!!.query<Any?>(org.mockito.kotlin.any<ContextQueryFunction<Any?>>())).thenThrow(RuntimeException::class.java)

    val check = healthCheckService!!.healthCheck()
    Assertions.assertFalse(check)
  }
}
